ext/pcre/php_pcre.c

```diff
@@ -1940,10 +1926,12 @@ static zend_string *php_pcre_replace_func_impl(pcre_cache_entry *pce, zend_strin
 
 			/* Use custom function to get replacement string and its length. */
 			eval_result = preg_do_repl_func(
-				fci, fcc, subject, offsets, subpat_names, num_subpats, count,
+				fcc, subject, offsets, subpat_names, num_subpats, count,
 				pcre2_get_mark(match_data), flags);
 
-			ZEND_ASSERT(eval_result);
+			if (UNEXPECTED(eval_result == NULL)) {
+				goto error;
```

Comment:
I'm not entirely sure this is right. This error path calls into `pcre_handle_exec_error(count)` where `count` represents an error code normally. However, in this call this is a real count and not an error code.
I was first thinking that you should probably set `PCRE_G(error_code)` yourself. However, perhaps not even that is necessary because we're throwing a userland exception anyway (which is not a pcre error itself).

Reply:
Right, I wasn't exactly sure myself here.